### PR TITLE
bump versions and fix some clippy warnings

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

**Review by iduartgomez — CHANGES_REQUESTED** [review]
Merge pending changes in stdlib and point to main head in this pr and we can merge

File: stdlib
[the remark this answers is not in the document]
Comment (al8n) [reply stdlib:1]: The commit in this PR is the latest commit of stdlib, so do we want to also bump versions in stdlib?

[the remark this answers is not in the document]
Comment (iduartgomez) [reply stdlib:1]: yes please

**Review by iduartgomez — APPROVED** [review]
Accepted stdlib or, merge that and then you this

**Pull request merged**